test/src/Stream/test_insertion_operator.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 Arduino.  All rights reserved.
+ */
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include <catch.hpp>
+
+#include <StreamMock.h>
+
+/**************************************************************************************
+ * TEST CODE
+ **************************************************************************************/
+
+TEST_CASE ("Testing 'Format' initialisation", "[Stream-insertion-operator-01]") {
+  Format<char> fmt {'a', 2};
+  REQUIRE(fmt.data == 'a');
+  REQUIRE(fmt.modifier == 2);
+}
+
+TEST_CASE ("Testing 'format' helper function", "[Stream-insertion-operator-02]") {
+  Format<char> fmt {format('a', 2)};
+  REQUIRE(fmt.data == 'a');
+  REQUIRE(fmt.modifier == 2);
+}
+
+TEST_CASE ("Testing basic insertion operator", "[Stream-insertion-operator-03]") {
+  StreamMock mock;
+  mock << 'a' << 12 << 'b' << 34;  // Note we cannot test C strings because `StreamMock` has its own << operator.
+  REQUIRE(mock.available() == 6);
```

Comment:
Please expands the tests to also check the data contained in the `StreamMock` object.

Reply:
Done that.